File: src/shared/data/block.js
const sha256Hash = require("../util");

module.exports = class Block {
    constructor(timestamp, data, previousHash = '') {
        this.previousHash = previousHash;
        this.timestamp = timestamp;
        this.data = data;

        this.hash = this.calculateHash();
    }

    calculateHash() {
        return sha256Hash(this.previousHash,this.timestamp,JSON.stringify(this.data));
    }
}
  
  

File: src/shared/data/block.py
from hashlib import sha256
import jsonpickle



class Block:

    def __init__(self, timestamp, data:any, previous_hash=None):
        self.timestamp = timestamp
        self.previous_hash = previous_hash
        self.data = data

        self.hash = self.calculate_hash()

    def calculate_hash(self):
        return sha256(jsonpickle.encode(self).encode('utf-8')).hexdigest()
